### ISSUE TITLE
Typescript ESM resolution broken

### ISSUE THREAD

**Issue by jbolotin** [knex/knex#4018]
# Environment

Knex version: 0.21.3+
Database + version: PG most recent
OS: macOS

# Bug

1. It seems as though 0.21.3 has broken module resolution if `knexfile` is a typescript file. Without digging around too much, I suspect it has something to do with [this fix](https://github.com/knex/knex/commit/cfce685775028d420525e82412e3207f17c0f2bd#diff-22d5d2b57c69d2e1ac054c4d0f744ff1L45) that went out in `0.21.3`

2. `Error: Cannot find module 'knexconfig'` -- error importing a module from `knexfile.ts`. This error occurs when running `knex --knexfile ./knexfile.ts migrate:make -x ts --esm` from the cli

**Comment by aMoniker** [comment]
I'm running into the same (or a similar) issue. I have `baseUrl: "."` in my `tsconfig.json` for a Next.js project, which is working fine in that context. However, when I run knex migrations or seeds, modules included from the base path won't resolve. Including via relative paths works fine, so I'm doing it that way for now.

**Comment by mercmobily** [comment]
You've hit a module resolution bug that was introduced when trying to fix another issue - classic unintended consequences! The  problem is that ts-node's module resolution doesn't align with Node's ESM resolution algorithm, so imports that should work end up failing.

This is exactly the kind of subtle bug that makes us want to move away from ts-node. In #5800, we're switching to tsx, which uses Node's native resolution algorithm and just adds TypeScript compilation on top, avoiding these mismatches entirely.

The immediate fix is to bypass our broken loader:

```
NODE_OPTIONS='--import tsx/esm' knex --knexfile ./knexfile.ts migrate:latest
```

Once #5800 is complete, module resolution will work exactly as you expect from any other Node.js TypeScript project.
